### PR TITLE
Add support for multivariate polynomial basis

### PR DESCRIPTION
**Pull request by dalg24**
This will be used in the moving least squares operator to build the multivariate Vandermonde matrix.

### CODE REVIEW

File: packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    operator()( Point const &p ) const;
+};
+
+// NOTE: For now relying on Point::operator[]( int i ) to access the coordinates
+// which make it possible to use various types such as DTK::Point or
+// Kokkos::Array<double, DIM>
+
+template <>
+template <typename Point>
+KOKKOS_INLINE_FUNCTION Kokkos::Array<double, 4>
```
Comment (aprokop) [inline packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp]: Shouldn't you replace this `4` with the appropriate `size()`?

File: packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+MultivariatePolynomialBasis<Linear, 3>::operator()( Point const &p ) const
+{
+    return {{1., p[0], p[1], p[2]}};
+}
+
+template <>
+template <typename Point>
+KOKKOS_INLINE_FUNCTION Kokkos::Array<double, 10>
```
Comment (aprokop) [inline packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp]: See comment above.

File: packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp (hunk 1 above)
Comment (dalg24) [reply packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp]: I would have to replace by `MultivariatePolynomialBasis<Linear, 3>::size()` so don't really want to.

Comment (dalg24) [reply packages/Meshfree/src/DTK_MultivariatePolynomialBasis.hpp]: Anyway, I did change it.

**Comment by dalg24** [comment]
@aprokop Do you reckon I should move explicit instantiations into .cpp file?

**Pull request merged**